tests/nnfw_api/src/CircleGen.h

```diff
@@ -54,6 +54,47 @@ class CircleGen
 public:
   using Shape = std::vector<int32_t>;
 
+  using SparseIndexVectorType = circle::SparseIndexVector;
+  using SparseDimensionType = circle::DimensionType;
+
+  struct SparseIndexVector
+  {
+    SparseIndexVector(){};
+    ~SparseIndexVector(){};
+    std::vector<uint16_t> u16;
+  };
+
+  struct DimMetaData
+  {
+    DimMetaData() = delete;
+    DimMetaData(SparseDimensionType format, std::vector<uint16_t> array_segments,
+                std::vector<uint16_t> array_indices)
+        : _format{format},
+          _array_segments_type(SparseIndexVectorType::SparseIndexVector_Uint16Vector),
+          _array_indices_type(SparseIndexVectorType::SparseIndexVector_Uint16Vector)
+    {
+      _array_segments.u16 = array_segments;
+      _array_indices.u16 = array_indices;
```

Comment:
```suggestion
        : _format{format},
          _array_segments_type(SparseIndexVectorType::SparseIndexVector_Uint16Vector),
          _array_indices_type(SparseIndexVectorType::SparseIndexVector_Uint16Vector),
          __array_segments{array_segments}, _array_indices{array_indices}
    {
```

Reply:
And, there are three kind of types in `SparseIndexVector` - INT32, U16, U8. Maybe you will just support only U16 type for convenience, won't you?

Reply:
@mhs4670go Yes, I will update if it become necessary. Just adding a field and function will work. However, it is not necessary and I don't want to spend my time. I guess it is unlikely to have a model that requires `Int32` index or segments, which means its sparse dimension has > 65535 elements. We may have `uint8`, when sparse dimension max index <= 256 or num_nonzero <= 256.